In [1]:
import numpy as np # linear algebra 
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
# stats models: regression fitting via formulas
import statsmodels.formula.api as smf
# stats models: regression fitting via matrices of regression design
import statsmodels.api as sm

# Output

In [2]:
df = pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/Labour.csv')
df

capital  labour     output       wage
0     2.606563     184   9.250759  43.080307
1     1.323237      91   3.664310  27.780016
2    22.093692     426  28.781516  44.467748
3    10.737851      72   4.124642  39.734710
4     1.161365      46   2.890150  34.650709
..         ...     ...        ...        ...
564   2.625403      20   1.424376  33.477545
565   1.276386      61   2.109048  26.300732
566   1.953869     117   6.241870  41.153979
567   1.318527      46   7.902237  66.720139
568   2.832391     161   6.804085  36.618435

[569 rows x 4 columns]

In [4]:
# специфицируем модель через формулу
output_eq1 = smf.ols(formula='np.log(output)~np.log(capital)+np.log(labour)', data=df).fit()
# Коэфициенты модели с округление
output_eq1.params.round(2)

Intercept         -1.71
np.log(capital)    0.21
np.log(labour)     0.71
dtype: float64

### Спецификация
$$
log(output)=-1.71+0.21*log(capital)+0.71*log(labour)
$$

### Интерпретация
1. При увеличении капитала на 1%, выпуск продукции увеличивается на 0.21% при прочих равных
2. При увеличении использования труда на 1%, выпуск продукции увеличивается на 0.71%, при прочих равных

In [5]:
# специфицируем модель через формулу
output_eq1 = smf.ols(formula='np.log(output)~np.log(capital)+np.log(labour)+np.log(wage)', data=df).fit()
# Коэфициенты модели с округление
output_eq1.params.round(3)

Intercept         -5.007
np.log(capital)    0.149
np.log(labour)     0.720
np.log(wage)       0.921
dtype: float64

$$
log(output)=-5.007+0.149*log(capital)+0.72*log(labour)+0.921*log(wage)
$$

### Спецификация
1. Если увеличить использование капитала на 1%, то выпуск увеличится на 0.149%
2. Если увеличить количество рабочих (рабочей силы), то выпуск увеличится на 0.921%

# Cost equation

In [6]:
df = pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/Electricity.csv')
df

cost        q        pl      sl      pk      sk      pf      sf
0     0.2130      8.0   6869.47  0.3291  64.945  0.4197  18.000  0.2512
1     3.0427    869.0   8372.96  0.1030  68.227  0.2913  21.067  0.6057
2     9.4059   1412.0   7960.90  0.0891  40.692  0.1567  41.530  0.7542
3     0.7606     65.0   8971.89  0.2802  41.243  0.1282  28.539  0.5916
4     2.2587    295.0   8218.40  0.1772  71.940  0.1623  39.200  0.6606
..       ...      ...       ...     ...     ...     ...     ...     ...
153   6.8293    946.6  10642.16  0.0883  43.600  0.1914  51.463  0.7203
154   3.7605    377.0   7432.24  0.2117  74.120  0.2274  33.436  0.5609
155   3.9822    391.0   5826.04  0.1926  78.288  0.0924  44.633  0.7151
156  30.1880   5317.0   9586.63  0.0845  78.008  0.2009  41.840  0.7147
157  67.8562  15220.0   6986.09  0.1451  74.025  0.2596  25.662  0.5945

[158 rows x 8 columns]

In [7]:
# специфицируем модель через формулу
output_eq1 = smf.ols(formula='np.log(cost)~np.log(q)', data=df).fit()
# Коэфициенты модели с округление
output_eq1.params.round(3)

Intercept   -3.841
np.log(q)    0.836
dtype: float64

$$
log(cost)=-3.841+0.836*log(q)
$$

### Интерпретация
1. При увеличении производства электроэнергии на 1%, то общие издержки увеличиваются в среднем на 0.836, при прочих равных

In [8]:
# специфицируем модель через формулу
output_eq1 = smf.ols(formula='np.log(cost)~np.log(q)+np.log(pl)+np.log(pf)+np.log(pk)', data=df).fit()
# Коэфициенты модели с округление
output_eq1.params.round(3)

Intercept    -7.472
np.log(q)     0.838
np.log(pl)    0.044
np.log(pf)    0.713
np.log(pk)    0.188
dtype: float64

$$
log(cost)=-7.472+0.838*log(q)+0.044*log(pl)+0.713*log(pf)+0.188*log(pk)
$$

### Спецификация
1. Если увеличить производство электроэнергии на 1%, то общие издержки за год в среднем увеличатся на 0.838%, при прочих равных
